File: Algorithms/701. Insert into a Binary Search Tree/kotlin/solution.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun insertIntoBST(root: TreeNode?, `val`: Int): TreeNode? {
        var node = root
        while (node != null) {
            if (`val` < node.`val`) {
                if (node.left == null) {
                    node.left = TreeNode(`val`)
                    return root
                }
                node = node.left
            } else {
                if (node.right == null) {
                    node.right = TreeNode(`val`)
                    return root
                }
                node = node.right
            }
        }

        return TreeNode(`val`)
    }
}

File: Algorithms/701. Insert into a Binary Search Tree/php/solution.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {
    /**
     * @param TreeNode $root
     * @param Integer $val
     * @return TreeNode
     */
    function insertIntoBST($root, $val) {
        $node = $root;
        while ($node) {
            if ($val > $node->val) {
                if (!$node->right) {
                    $node->right = new TreeNode($val);
                    return $root;
                }
                $node = $node->right;
            } else {
                if (!$node->left) {
                    $node->left = new TreeNode($val);
                    return $root;
                }
                $node = $node->left;
            }
        }
        return new TreeNode($val);
    }
}
